src/software/ai/passing/cost_function.cpp

```diff
@@ -332,3 +356,24 @@ double getStaticPositionQuality(const Field& field, const Point& position,
 
     return on_field_quality * near_friendly_goal_quality * in_enemy_defense_area_quality;
 }
+
+
+Duration estimateBallTimeToPosition(const Ball& ball, const Point& pos, double ball_decel) {
+    /**
+     * Using kinematics to roughly model ball decel, since we have a tendency
+     * to take long passes where the ball slows down a lot and enemies can intercept
+     *
+     * vf^2 = vi^2 + 2*a*d
+     * vf = vi + a*t
+     *
+     * t = (sqrt(vi^2 + 2ad) -vi) / a
+     */
+
+    double dist = distance(pos, ball.position());
```

Comment:
Was originally planning on trying to add a simple ball friction model to account for how much the ball slowed down over long passes. Was hoping this would help prevent interceptions of long passes.